services/oc-pmrmrg/src/pmr_mrg/run_model.py

```diff
@@ -0,0 +1,96 @@
+import OpenCOR
+import sys
+import json
+import csv
+
+from pathlib import Path
+
+
+# modelpath = "/home/zhuang/Downloads/guyton_antidiuretic_hormone_2008.cellml"
+inputjson= str(sys.argv[1])
+modelpath = str(sys.argv[2])
+mapfile = str(sys.argv[3])
+
+# parse user inputs from json file
+inputpath = Path(inputjson)
+with inputpath.open("r") as fp:
+    inputdata_unmapped = json.load(fp)
+
+mapfilepath = Path(mapfile)
+with mapfilepath.open("r") as fp:
+    inputkeymap = json.load(fp)
+inputdata = {newkey: inputdata_unmapped[oldkey] for (oldkey, newkey) in inputkeymap.items()}
+print(inputdata)
+
+starttime = float(inputdata["starttime"])
+endtime = float(inputdata["endtime"])
+timeincr = float(inputdata["timeincr"])
+
+# some basic verification
+if modelpath.endswith('cellml') or modelpath.endswith('sedml'):
+    model = OpenCOR.openSimulation(modelpath)
+else:
+    sys.exit('Invalid file type: only .cellml and .sedml accepted')
+
+
+if endtime<starttime:
+    print('ending time must be greater than starting time, using 1000s after start instead')
+    endtime = starttime +1000
```

Comment:
as a comment, we should improve the docker labels so that this kind of validation may be done before the service starts

Reply:
until that's in place I will keep these in there